tests/test_widgets.py

```diff
@@ -1059,6 +1059,38 @@ def f(x: Set[Lit]): ...
     assert sel.choices == get_args(Lit)
 
 
+def test_separator_singleton():
+    from magicgui.types import Separator, _Separator
+
+    sep1 = Separator
+    sep2 = _Separator()
+    assert sep1 is sep2
+
+
+def test_separator():
+    from magicgui.types import Separator
+
+    sep = [[Separator] * (i + 1) for i in range(4)]
+    a = [1, 2, *sep[0], 4, *sep[1], 6, 7, *sep[2], 9, *sep[3], 11, 12, *sep[0], 14, *sep[0]]
+    b = [1, 2, *sep[0], 4, *sep[1], 6, 7, *sep[2], 9, *sep[3], 6, 7, *sep[0], 9, *sep[0]]
+    a2 = (1, 2, None, 4, None, None, 6, 7, None, None, None, 9, None, None, None, None, 11, 12, None, 14, None)
+    b2 = (1, 2, None, 4, None, None, 6, 7, None, None, None, 9, None, None, None, None, None, None)
+
+    @magicgui(
+        call_button="Test Separator",
+        a={"widget_type": "ComboBox", "choices": a},
+        b={"widget_type": "ComboBox", "choices": b},
+        result_widget=True,
+    )
+    def widget_with_separator(a=a[0], b=b[0]):
+        return
+
+    assert widget_with_separator.a.native.count() == 21
+    assert widget_with_separator.b.native.count() == 18
+    assert widget_with_separator.a.choices == a2
+    assert widget_with_separator.b.choices == b2
```

Comment:
sorry, I missed this the first time.  I think this is a bit unfortunate actually.  Since the separator is a purely UI element (i.e. it can't actually be selected), I'm not sure it should be returned in the choices.  I think I would expect

```python
assert widget_with_separator.a.choices == (1, 2, 4, 6, 7, 9, 11, 12, 14)
```

thoughts?

Reply:
one way to accomplish this would be to `.setItemData(i, '<some-special-key>')` immediately after calling `insertSeparator`, and then filtering out those things in _mgui_get_choices

Reply:
Absolutely, separators are items in Qt/QComboBox, so we can't avoid having them, and your `.setItemData(i, '<some-special-key>')` is very clever. You are also right about not returning all the `None`s 🤦🏻‍♂️I'll update `_mgui_get_choices()` and `_mgui_get_count()`.